src/lints/clone_on_copy.rs

```diff
@@ -65,3 +81,95 @@ fn check_clone_usage(
         });
     }
 }
+
+fn fix_clone_on_copy(db: &dyn SemanticGroup, node: SyntaxNode) -> Option<(SyntaxNode, String)> {
+    let ast_expr_binary = ast::ExprBinary::cast(db.upcast(), node)?;
+
+    let module_file_id = helper::find_module_file_containing_node(db, &node)?;
+
+    let ast_expr = ast_expr_binary.lhs(db.upcast());
+
+    let expr_semantic = get_expr_semantic(db, module_file_id, &ast_expr, &ast_expr_binary)
+        .expect("Failed to find expression semantic.");
+
+    // Extract the number of `@` snapshots from the type.
+    // Each `@` will later be represented as a `*` prefix in the output.
+    let (mut snapshot_count, _) = peel_snapshots(db, expr_semantic.ty());
+
+    // `clone(self: @T)` expects an `@`, so the compiler will automatically insert
+    // an `@` into the type if it was not explicitly provided by the user.
+    // In such cases, the expression will be of type `Expr::Snapshot`,
+    // meaning that `peel_snapshots` would count one coercion too many.
+
+    // However, if the `@` was explicitly written by the user,
+    // the expression will be of another type, such as `Expr::Var`,
+    // and `peel_snapshots` will have already counted the correct number of `@`.
+
+    // Therefore, we need to manually subtract one from the snapshot count
+    // when the expression is a `Expr::Snapshot` to correct this.
```

Comment:
nit: make it a single comment so others are sure it related to the `if let Expr::Snapshot(_) = expr_semantic {` line.
```suggestion
    // `clone(self: @T)` expects an `@`, so the compiler will automatically insert
    // an `@` into the type if it was not explicitly provided by the user.
    // In such cases, the expression will be of type `Expr::Snapshot`,
    // meaning that `peel_snapshots` would count one coercion too many.
    //
    // However, if the `@` was explicitly written by the user,
    // the expression will be of another type, such as `Expr::Var`,
    // and `peel_snapshots` will have already counted the correct number of `@`.
    //
    // Therefore, we need to manually subtract one from the snapshot count
    // when the expression is a `Expr::Snapshot` to correct this.
```